adala/runtimes/_litellm.py

```diff
@@ -37,32 +80,43 @@ class LiteLLMChatRuntime(LiteLLMInferenceSettings, Runtime):
 
     model_config = ConfigDict(arbitrary_types_allowed=True)  # for @computed_field
 
+    def as_inference_settings(self, **override_kwargs) -> LiteLLMInferenceSettings:
+        inference_settings = LiteLLMInferenceSettings(
+            **self.dict(include=LiteLLMInferenceSettings.model_fields.keys())
+        )
+        inference_settings.update(**override_kwargs)
+        return inference_settings
+
     def init_runtime(self) -> "Runtime":
         # check model availability
+        # extension of litellm.check_valid_key for non-openai deployments
         try:
-            if self.api_key:
-                litellm.check_valid_key(model=self.model, api_key=self.api_key)
-        except NotFoundError:
+            messages = [{"role": "user", "content": "Hey, how's it going?"}]
+            litellm.completion(
+                messages=messages, **self.as_inference_settings(max_tokens=10).dict()
+            )
+        except AuthenticationError:
+            raise ValueError(
+                f'Requested model "{self.model}" is not available with your api_key and settings.'
+            )
+        except Exception as e:
             raise ValueError(
-                f'Requested model "{self.model}" is not available with your api_key.'
+                f'Failed to check availability of requested model "{self.model}": {e}'
             )
         return self
 
     def get_llm_response(self, messages: List[Dict[str, str]]) -> str:
```

Comment:
I realized this too late in the last CR, but can we please rename this func? It's really confusing that we import get_llm_response and then define a new, separate, get_llm_response

Reply:
`self.get_llm_response` is a legacy function used in `Agent.learn()`, so it's not changeable without more refactoring. The new `*_get_llm_response` utils in `llm.py` are the ones I just removed, so no more name collision.